### PR TITLE
Some doctests and examples and a missing "."

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Comment by ViralBShah** [comment]
Rebased on master.

File: base/file.jl [hunk 1]
```diff
@@ -35,6 +35,17 @@ import .Base.RefValue
     pwd() -> AbstractString
 
 Get the current working directory.
+
+# Examples
+```julia-repl
+julia> pwd()
+"/home/JuliaUser/.julia/v0.7"
```
Comment (fredrikekre) [inline base/file.jl]: The `v0.7` folder is not used anymore with Pkg(3). Just change to `"/home/JuliaUser"` maybe?

File: base/file.jl [hunk 2]
```diff
@@ -80,6 +104,18 @@ end
     cd(f::Function, dir::AbstractString=homedir())
 
 Temporarily changes the current working directory and applies function `f` before returning.
+
+# Examples
+```jldoctest
+julia> cd(readdir, joinpath(Sys.BINDIR, Base.DATAROOTDIR, "julia"))
```
Comment (fredrikekre) [inline base/file.jl]: Perhaps overly complicated to use `Sys.BINDIR` and `Base.DATAROOTDIR` just to show how to use `cd`? It takes focus on what we are trying to show.

File: base/file.jl [hunk 3]
```diff
@@ -461,6 +569,29 @@ end
     readdir(dir::AbstractString=".") -> Vector{String}
 
 Return the files and directories in the directory `dir` (or the current working directory if not given).
+
+# Examples
+```jldoctest
+julia> readdir(joinpath(Sys.BINDIR, Base.DATAROOTDIR))
```
Comment (fredrikekre) [inline base/file.jl]: Same here, better with an example without `Sys.BINDIR` and `Base.DATAROOTDIR` IMO.

File: base/file.jl [hunk 4]
```diff
@@ -277,6 +369,22 @@ end
...
 Return `path`.
+
+# Examples
+```julia-repl
+julia> write("my_little_file", 2);
+
+julia> mtime("my_little_file")
+1.5273815391135583e9
+
+julia> touch("my_little_file")
```
Comment (fredrikekre) [inline base/file.jl]: Append `;` ?

File: base/file.jl [hunk 5]
```diff
@@ -99,6 +135,17 @@ directory. If the directory already exists, or some intermediate directories do
 this function throws an error. See [`mkpath`](@ref) for a function which creates all
 required intermediate directories.
 Return `path`.
+
+# Examples
+```julia-repl
+julia> mkdir("faketestingdir")
```
Comment (fredrikekre) [inline base/file.jl]: Whats "fake" with this directory? Perhaps just `testdir` would be enough? (And same for the `mkpath` example below.)

**Review by fredrikekre — COMMENTED** [review]
Closes #18827 ?

File: base/file.jl (hunk 2 above)
Comment (kshyatt) [reply base/file.jl]: Yeah, I wanted something that would work as a doctest but maybe a `julia-repl` is ok for now

**Pull request merged**